### PR TITLE
Test the data processing

### PR DESCRIPTION
**Pull request by SPalominos**
Test the preprocessing and postprocessing classes registered in the class DataProcessingManager. For each, test the well processing in case of good configuration and test the error detection in case of wrong configuration.
Part of the issue #1200.

### CODE REVIEW

File: bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+     * Test the preprocessing of an incoming BoundingBoxData and if the geometry generated is correct.
+     * Then check its postprocessing and if its string representation is correct.
+     * @throws MalformedScriptException
+     */
+    @Test
+    public void testBoundingBoxProcessing() throws MalformedScriptException {
+        //Store into the dataMap the boundingBox data
+        URI uri = URI.create(UUID.randomUUID().toString());
+        Map<URI, Object> dataMap = new HashMap<>();
+        dataMap.put(uri, "EPSG:4326;0,0,1,1");
```
Comment (ebocher) [inline bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java]: Is this string switchable ? EPSG:4326;0,0,1,1 -> 0,0,1,1; EPSG:4326

File: bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        //Store into the dataMap the geometryData data
+        URI uri = URI.create(UUID.randomUUID().toString());
+        Map<URI, Object> dataMap = new HashMap<>();
+        dataMap.put(uri, "POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))");
+        //Generate the InputDescriptionType object containing the GeometryData
+        List<Format> formatList = FormatFactory.getFormatsFromExtensions(new String[]{".txt"});
+        List<DataType> geometryList = new ArrayList<>();
+        geometryList.add(DataType.POLYGON);
+        GeometryData geometryData = new GeometryData(formatList, geometryList);
+        geometryData.setExcludedTypeList(new ArrayList<DataType>());
```
Comment (ebocher) [inline bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java]: What about if geometryData.setExcludedTypeList is no set ?

File: bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        Assert.assertEquals("The bounding box geometry wasn't the one expected.",
+                "POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))", str);
+    }
+
+    /**
+     * Test the preprocessing in the different cases of wrong GeometryData configuration (bad geometry, bad dimension ...)
+     * @throws MalformedScriptException
+     */
+    @Test
+    public void testBadGeometryProcessing() throws MalformedScriptException {
```
Comment (ebocher) [inline bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java]: I think you must add you unit test to follow this practice 

@Test(expected = MalformedScriptException.class)
 public void testBadGeometryProcessing() throws Throwable {

see here an example https://github.com/orbisgis/h2gis/blob/4ba5249a4a229c38d30323a152c1d8d5524fc776/h2gis-functions/src/test/java/org/h2gis/functions/spatial/SpatialFunctionTest.java#L176

File: bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java (hunk 1 above)
Comment (SPalominos) [inline bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java]: No, but I can update the code to do it if needed.

Comment (ebocher) [inline bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java]: Yes please

File: bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java (hunk 2 above)
Comment (SPalominos) [inline bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java]: If not set, it means that no types are excluded.

File: bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java (hunk 3 above)
Comment (SPalominos) [inline bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java]: I'll fix it in the incoming commit

File: bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java (hunk 2 above)
Comment (ebocher) [inline bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java]: so it is not necessary to add geometryData.setExcludedTypeList(new ArrayList<DataType>());

File: bundles/wps/wps-service/src/main/java/org/orbisgis/wpsservice/controller/execution/BoundingBoxProcessing.java [hunk 4]
```diff
@@ -82,7 +82,18 @@
...
+                String srid;
+                if(split[0].contains(":")){
+                    srid = str.split(";")[0].split(":")[1];
+                    wkt = str.split(";")[1].split(",");
+                }
+                else{
+                    srid = str.split(";")[1].split(":")[1];
+                    wkt = str.split(";")[0].split(",");
+                }
+
```
Comment (SPalominos) [inline bundles/wps/wps-service/src/main/java/org/orbisgis/wpsservice/controller/execution/BoundingBoxProcessing.java]: Test add with the last commit.

**Pull request merged**